assets/js/modules/analytics/datastore/setup-flow.test.js

```diff
@@ -0,0 +1,231 @@
+		[
+			{
+				_id: '2000',
+				_propertyID: '1000',
+				name: 'properties/1000/webDataStreams/2000',
+				// eslint-disable-next-line sitekit/acronym-case
+				measurementId: '1A2BCD345E',
+				// eslint-disable-next-line sitekit/acronym-case
+				firebaseAppId: '',
+				createTime: '2014-10-02T15:01:23Z',
+				updateTime: '2014-10-02T15:01:23Z',
+				defaultUri: 'http://example.com',
+				displayName: 'Test GA4 WebDataStream',
+			},
+		],
+		{ propertyID: 'foobar' }
+	);
+};
+
+describe( 'modules/analytics setup-flow', () => {
+	let registry;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		// Receive empty settings to prevent unexpected fetch by resolver.
+		registry.dispatch( STORE_NAME ).receiveGetSettings( {} );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getSetupFlowMode', () => {
+			it( 'returns “legacy” if the modules/analytics-4 store is not available ', async () => {
+				// just for this test create new registry with only this module
+				const newRegistry = createRegistry();
+
+				[
+					modulesAnalytics,
+				].forEach( ( { registerStore } ) => registerStore?.( newRegistry ) );
+
+				registry = newRegistry;
+				// Receive empty settings to prevent unexpected fetch by resolver.
+				registry.dispatch( STORE_NAME ).receiveGetSettings( {} );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( 'legacy' );
+			} );
+
+			it( 'should return “legacy” if isAdminAPIWorking() returns false ', () => {
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveError(
+					new Error( 'foo' ), 'getProperties', [ 'foo', 'bar' ]
+				);
+
+				expect( registry.select( MODULES_ANALYTICS_4 ).isAdminAPIWorking() ).toBe( false );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( 'legacy' );
+			} );
+
+			it( 'should return undefined if isAdminAPIWorking() returns undefined ', () => {
+				expect( registry.select( MODULES_ANALYTICS_4 ).isAdminAPIWorking() ).toBe( undefined );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( undefined );
+			} );
+
+			it( 'should return “ua” if there is no account selected', () => {
+				expect( registry.select( STORE_NAME ).getAccountID( accountID ) ).toBe( undefined );
+
+				populateAnalytics4DataStore( registry );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( 'ua' );
+			} );
+
+			it( 'should return “ua” if selected account returns an empty array from GA4 getProperties selector', () => {
+				registry.dispatch( STORE_NAME ).setAccountID( accountID );
+				populateAnalyticsDataStore( registry );
+
+				// as per spec https://github.com/google/site-kit-wp/issues/3169
+				// For isAdminAPIWorking() to return true:
+				// "The state['properties'] object has at least one account with a non-empty array of properties;"
```

Comment:
Please start sentences in comments with a capital letter 😄 
```suggestion
				// For isAdminAPIWorking() to return true:
				// "The state['properties'] object has at least one account with a non-empty array of properties;"
				// See: https://github.com/google/site-kit-wp/issues/3169
```